packages/webpack-plugin/src/plugin-utils.ts

```diff
@@ -0,0 +1,368 @@
+    if (match) {
+        return publicPath + match[1];
+    }
+    match = source.toString().match(/module.exports\s*=\s*"(.*?)"/);
+    if (match) {
+        return match[1];
+    }
+    match = source.toString().match(/export\s+default\s+"(.*?)"/);
+    if (match) {
+        return match[1];
+    }
+    throw new Error('unknown asset module format ' + source);
+}
+
+export function extractDataUrlFromAssetModuleSource(source: string): string {
+    let match = source.match(/.exports\s*=\s*"(.*?)"/);
+    if (match) {
+        return match[1];
+    }
+    match = source.toString().match(/export\s+default\s+"(.*?)"/);
+    if (match) {
+        return match[1];
+    }
+    throw new Error('unknown data url asset module format ' + source);
+}
+
+type AssetNormalModule = NormalModule & { loaders: [{ loader: 'file-loader' | 'url-loader' }] };
+
+export function isLoadedWithKnownAssetLoader(module: Module): module is AssetNormalModule {
+    if (module instanceof NormalModule) {
+        return !!module.loaders.find(({ loader }) =>
+            /[\\/](file-loader)|(url-loader)[\\/]/.test(loader)
+        );
+    }
+    return false;
+}
+
+export function outputOptionsAwareHashContent(
+    createHash: webpackCreateHash,
+    outputOptions: webpackOutputOptions,
+    content: string
+) {
+    const hash = createHash(outputOptions.hashFunction || 'md4');
+    if (outputOptions.hashSalt) {
+        hash.update(outputOptions.hashSalt);
+    }
+
+    hash.update(content);
+    const fullHash = /** @type {string} */ hash.digest(outputOptions.hashDigest);
+    const contentHash = fullHash.slice(0, outputOptions.hashDigestLength);
+    return contentHash.toString();
+}
+
+export function injectLoader(compiler: Compiler) {
+    if (!compiler.options.module.rules) {
+        compiler.options.module.rules = [];
+    }
+    compiler.options.module.rules.unshift({
+        test: /\.st\.css$/,
+        loader: require.resolve(join(__dirname, 'loader')),
+        sideEffects: true,
+    });
+}
+
+export function createDecacheRequire(compiler: Compiler) {
+    if (compiler.watchMode) {
+        const cacheIds = new Set<string>();
+        compiler.hooks.done.tap('decache require', () => {
+            for (const id of cacheIds) {
+                decache(id);
+            }
+            cacheIds.clear();
+        });
+        return (id: string) => {
+            cacheIds.add(id);
+            return require(id);
+        };
+    } else {
+        return require;
+    }
+}
+
+export function createStaticCSS(
+    staticPublicPath: string,
+    stylableModules: Set<Module>,
+    assetsModules: Map<string, NormalModule>,
+
+    chunkGraph: ChunkGraph,
+    moduleGraph: ModuleGraph,
+    runtime: string,
+    runtimeTemplate: RuntimeTemplate,
+    dependencyTemplates: DependencyTemplates
+) {
+    const cssChunks = Array.from(stylableModules)
+        .filter((m) => m.buildMeta.stylable.isUsed !== false)
```

Comment:
remove the !== false? is there a situation where it can be undefined?

Reply:
yes. it is represented in the type but I didn't used the helper to get the metadata so you couldn't see it. will use it now